### PR TITLE
Add `TextClassification`, `UMAP`, `DBSCAN` and `TextClustering` tasks

### PR DESCRIPTION
**Pull request by plaguss**
## Description

This PR adds a general `TextClassification` task, which can be useful for both label and multilabel classification.

Also `UMAP`, `DBSCAN` and `TextClustering`, which are quite related to each other, to generate clusters of text.

It was defined as part of a pipeline for text clustering, so may be biased towards that definition, but should be general enough to work on any common text classification tasks.

**TODO**. Create an example in the docs of a full text clustering pipeline, including the label inference.

### CODE REVIEW

File: src/distilabel/steps/clustering/dbscan.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    StepInput,
+)
+
+if TYPE_CHECKING:
+    from sklearn.cluster import DBSCAN as _DBSCAN
+
+    from distilabel.steps.typing import StepOutput
+
+
+class DBSCAN(GlobalStep):
```
Comment (gabrielmbmb) [inline src/distilabel/steps/clustering/dbscan.py:33]: Lets add the `unload` method that set the `self._clusterer = None`

File: src/distilabel/steps/clustering/umap.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    StepInput,
+)
+
+if TYPE_CHECKING:
+    from umap import UMAP as _UMAP
+
+    from distilabel.steps.typing import StepOutput
+
+
+class UMAP(GlobalStep):
```
Comment (gabrielmbmb) [inline src/distilabel/steps/clustering/umap.py:33]: Add `unload` method to set `self._umap = None`

**Review by gabrielmbmb — CHANGES_REQUESTED** [review]
LGTM!
